pkg/egressgateway/policy.go

```diff
@@ -81,12 +82,32 @@ func (config *PolicyConfig) matchesEndpointLabels(endpointInfo *endpointMetadata
 	return false
 }
 
+// matchesNodetLabels determines if the given node lables is a match for the
+// policy config based on matching labels.
+func (config *PolicyConfig) matchesNodetLabels(nodeLabels map[string]string) bool {
+	if len(config.nodeSelectors) == 0 {
+		return true
+	}
+	labelsToMatch := k8sLabels.Set(nodeLabels)
+	for _, selector := range config.nodeSelectors {
+		if selector.Matches(labelsToMatch) {
+			return true
+		}
+	}
+	return false
+}
+
 // updateMatchedEndpointIDs update the policy's cache of matched endpoint IDs
-func (config *PolicyConfig) updateMatchedEndpointIDs(epDataStore map[endpointID]*endpointMetadata) {
+func (config *PolicyConfig) updateMatchedEndpointIDs(epDataStore map[endpointID]*endpointMetadata, nodes []nodeTypes.Node) {
 	config.matchedEndpoints = make(map[endpointID]*endpointMetadata)
-
+	nodesAddresses2Labels := make(map[string]map[string]string)
```

Comment:
Don't think manager.nodes can change in this section, so it seems like we're doing extra work by recomputing this for each endpoint?